src/Resources/Pieces.php

```diff
@@ -30,29 +27,35 @@ public function __construct(array $attributes = [])
      * @param  \Mvdnbrk\DhlParcel\Resources\Piece|array  $value
      * @return void
      */
-    public function add($value)
+    public function add($value): void
     {
         if ($value instanceof Piece) {
             $this->items[] = $value;
 
             return;
         }
 
-        $this->items[] = new Piece($value);
+        if (is_array($value)) {
+            $this->items[] = new Piece($value);
+
+            return;
+        }
+
+        throw ResourceNotAccepted::forResource($value);
     }
 
     /**
      * Convert the Pieces resource to an array.
      *
      * @return array
      */
-    public function toArray()
+    public function toArray(): array
     {
```

Comment:
Please remove `: array` for same reason as above. If you would like to change this do this in a separate PR.